pkg/project/registry/project/proxy/proxy.go

```diff
@@ -128,6 +128,29 @@ func (s *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, err
 	return convertNamespace(namespace), nil
 }
 
+func (s *REST) Update(ctx kapi.Context, obj runtime.Object) (runtime.Object, bool, error) {
+	project, ok := obj.(*api.Project)
+	if !ok {
+		return nil, false, fmt.Errorf("not a project: %#v", obj)
+	}
+
+	oldObj, err := s.Get(ctx, project.Name)
+	if err != nil {
+		return nil, false, err
+	}
+	s.updateStrategy.PrepareForUpdate(obj, oldObj)
+	if errs := s.updateStrategy.ValidateUpdate(ctx, obj, oldObj); len(errs) > 0 {
+		return nil, false, kerrors.NewInvalid("project", project.Name, errs)
+	}
+
+	namespace, err := s.client.Update(convertProject(project))
```

Comment:
> convertProject feels fragile to use for an update, since it builds a new namespace from scratch, copying over specific fields. We need to make sure a namespace with every field populated retains all those fields when updated as a project. That feels like some sort of fuzz test (create fully populated namespace, update as project, fetch the new namespace, ensure the only difference is the resource version)

If convertProject doesn't convert the project to a namespace, what does it do?  You're saying you want better/more tests on the casting?